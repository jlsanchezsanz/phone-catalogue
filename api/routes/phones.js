const phonesRoutes = (app, fs) => {
  const dataPath = './data/phones.json';

  app.get('/phones', (_, res) => {
    fs.readFile(dataPath, 'utf8', (err, data) => {
      setTimeout(() => {
        if (err) {
          throw err;
        }
        res.setHeader('Access-Control-Allow-Origin', '*');
        res.send(JSON.parse(data));
      }, 3000);
    });
  });
};

module.exports = phonesRoutes;
