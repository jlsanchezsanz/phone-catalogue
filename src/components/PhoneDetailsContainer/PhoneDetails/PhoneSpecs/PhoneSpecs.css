.specs {
  display: flex;
  flex-direction: column;
}


dl {
  margin: 0;
  padding: 20px 30px;
  text-align: left;
}

dt {
  width: 100px;
}

dd {
  color: #333;
  font-weight: 300;
}

dt,
dd {
  display: inline-block;
}
