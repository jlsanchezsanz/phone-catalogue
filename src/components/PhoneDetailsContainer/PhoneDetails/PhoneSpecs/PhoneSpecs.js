import React from 'react';
import PropTypes from 'prop-types';

import { PhonePropType } from '../../../../prop-types/phone.prop-type';
import './PhoneSpecs.css';

export default function PhoneSpecs(props) {
  return (
    <div className='specs'>
      <h4>SPECS</h4>
      <ul className='list-group'>
        <li className='list-group-item' key='screen'>
          <dl>
            <dt>Screen:</dt>
            <dd>{props.phone.screen}</dd>
          </dl>
        </li>
        <li className='list-group-item' key='processor'>
          <dl>
            <dt>Processor:</dt>
            <dd>{props.phone.processor}</dd>
          </dl>
        </li>
        <li className='list-group-item' key='ram'>
          <dl>
            <dt>Memory:</dt>
            <dd>{props.phone.ram}GB RAM</dd>
          </dl>
        </li>
        <li className='list-group-item' key='description'>
          <dl>
            <dt>Description:</dt>
            <dd>{props.phone.description}</dd>
          </dl>
        </li>
      </ul>
    </div>
  );
}

PhoneSpecs.propTypes = {
  phone: PropTypes.shape(PhonePropType).isRequired
};
