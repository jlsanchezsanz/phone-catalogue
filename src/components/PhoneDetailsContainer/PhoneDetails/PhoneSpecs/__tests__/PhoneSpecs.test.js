import React from 'react';
import { shallow } from 'enzyme';

import PhoneSpecs from '../PhoneSpecs';
import { phonesMock } from '../../../../../mocks/phones.mock';

const setUpShallow = (phone) => {
  const component = shallow(<PhoneSpecs phone={phone} />);
  return component;
};

describe('PhoneSpecs', () => {
  let component;

  it('should display phone specs', () => {
    component = setUpShallow(phonesMock.data[0]);
    expect(component).toMatchSnapshot();
  });
});
