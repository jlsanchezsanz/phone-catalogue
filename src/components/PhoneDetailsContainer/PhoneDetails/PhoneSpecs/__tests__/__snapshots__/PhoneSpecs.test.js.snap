// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneSpecs should display phone specs 1`] = `
<div
  className="specs"
>
  <h4>
    SPECS
  </h4>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
      key="screen"
    >
      <dl>
        <dt>
          Screen:
        </dt>
        <dd>
          4,7 inch IPS
        </dd>
      </dl>
    </li>
    <li
      className="list-group-item"
      key="processor"
    >
      <dl>
        <dt>
          Processor:
        </dt>
        <dd>
          A10 Fusion
        </dd>
      </dl>
    </li>
    <li
      className="list-group-item"
      key="ram"
    >
      <dl>
        <dt>
          Memory:
        </dt>
        <dd>
          2
          GB RAM
        </dd>
      </dl>
    </li>
    <li
      className="list-group-item"
      key="description"
    >
      <dl>
        <dt>
          Description:
        </dt>
        <dd>
          iPhone 7 dramatically improves the most important aspects of the iPhone experience. It introduces advanced new camera systems. The best performance and battery life ever in an iPhone. Immersive stereo speakers. The brightest, most colorful iPhone display. Splash and water resistance*. And it looks every bit as powerful as it is. This is iPhone 7.
        </dd>
      </dl>
    </li>
  </ul>
</div>
`;
